test/unit/linear_algebra/test_lapacke.cpp

```diff
@@ -0,0 +1,138 @@
+#include <tuvx/linear_algebra/linear_algebra.hpp>
+
+#include <gtest/gtest.h>
+
+#include <cfloat>
+#include <chrono>
+#include <cstddef>
+#include <cstdlib>
+#include <cstring>
+#include <ctime>
+#include <fstream>
+#include <lapacke.h>
```

Comment:
I think this works for GNU, but not Intel compiler that uses MKL. Is there an easy way to switch between compilers?

Reply:
For MKL, we have to `#include<mkl_lapacke.h>`

Reply:
yeah, so I am looking for a setup that can easily switch between these two header files depending on the chosen compilers, rather than changing them manually.

Reply:
> yeah, so I am looking for a setup that can easily switch between these two header files depending on the chosen compilers, rather than changing them manually.

I have edited the CMAKE file to define a preprocessor flag based on the compiler used. The code will include `<mkl_lapacke.h>` or `<lapacke.h>` accordingly.